### PR TITLE
feat: Add home and video listing page

### PR DESCRIPTION
**Pull request by jalajcodes**
Live: https://max-video.herokuapp.com

### CODE REVIEW

File: frontend/src/components/Icons.js [hunk 1]
```diff
@@ -10,6 +10,19 @@ export const LogoIcon = (props) => (
...
 );
 
+export const MenuIcon = (props) => (
+  <svg
+    height="22"
+    width="22"
+    fill="#AAAAAA"
+    id="icon-dots-three-vertical"
+    viewBox="0 0 20 20"
+    onClick={props.onClick}
```
Comment (nayyyhaa) [inline frontend/src/components/Icons.js]: try to destructure props

File: frontend/src/components/VideoCard.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import useClickOutside from "../hooks/useClickOutside";
+import Wrapper from "../styles/VideoCard";
+import { IMAGE_BASE_URL, POSTER_SIZE } from "../utils/tmdb";
+import { DislikeIcon, LikeIcon, MenuIcon, SubIcon } from "./Icons";
+
+const VideoCard = ({ details }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const ref = useRef();
+
+  useClickOutside(ref, () => setIsOpen(false));
```
Comment (nayyyhaa) [inline frontend/src/components/VideoCard.js:12]: appreciation: good use of custom-hooks here.

File: frontend/src/hooks/useClickOutside.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      if (!ref.current || ref.current.contains(event.target)) {
+        return;
+      }
+      handler(event);
+    };
+    document.addEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
```
Comment (nayyyhaa) [inline frontend/src/hooks/useClickOutside.js:15]: try not to use `addEventListener` in react. read handling event from here: https://reactjs.org/docs/handling-events.html

File: frontend/src/pages/Home.js [hunk 4]
```diff
@@ -1,7 +1,53 @@
...
+          </div>
+        </Link>
+        <Link to="/feed/videos?genre=10749">
+          <div className="category-card">
+            <span>Romance</span>
+          </div>
+        </Link>
+        <Link to="/feed/videos?genre=878">
+          <div className="category-card">
+            <span>Sci Fi</span>
```
Comment (nayyyhaa) [inline frontend/src/pages/Home.js]: suggestion: you can crate a data array & can map it here for better readability.

File: frontend/src/pages/Videos.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  const params = Object.fromEntries(new URLSearchParams(searchParams));
+  const selectedGenre = useRef(params.genre ? params.genre : "popular");
+
+  const { data, isFetching, refetch } = useQuery(
+    `movies`,
+    () => fetchMovies(selectedGenre.current),
+    {
+      staleTime: Infinity,
+    }
+  );
```
Comment (nayyyhaa) [inline frontend/src/pages/Videos.js:20]: woah! learned implementation of `useQuery` here. nice!

**Review by nayyyhaa — APPROVED** [review]
I've given a few suggestions but nice work overall. Code is well written& easily understandable.

File: frontend/src/hooks/useClickOutside.js (hunk 3 above)
Comment (jalajcodes) [reply frontend/src/hooks/useClickOutside.js:15]: Its fine I think, I have seen this pattern used in production, also removing listener on unmount can't be done that way, so I am doing it like this.

File: frontend/src/pages/Home.js (hunk 4 above)
Comment (jalajcodes) [reply frontend/src/pages/Home.js]: Thanks for pointing out, will do that

File: frontend/src/pages/Videos.js (hunk 5 above)
Comment (jalajcodes) [reply frontend/src/pages/Videos.js:20]: react-query is awesome😇

File: frontend/src/components/Icons.js (hunk 1 above)
Comment (jalajcodes) [reply frontend/src/components/Icons.js]: will do that

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me! Just make the changes suggested by Neha

**Pull request merged**